src/main/java/io/github/thebusybiscuit/slimefun4/implementation/tasks/RadioactivityTask.java

```diff
@@ -0,0 +1,175 @@
+
+        if (!profile.hasFullProtectionAgainst(ProtectionType.RADIATION)) {
+            for (ItemStack item : p.getInventory()) {
+                if (item == null || item.getType() == Material.AIR) continue;
+                ItemStack tmpItem = item;
+
+                if (!(item instanceof SlimefunItemStack) && radioactiveItems.size() > 1) {
+                    // Performance optimization to reduce ItemMeta calls
+                    tmpItem = ItemStackWrapper.wrap(tmpItem);
+                }
+                
+                for (SlimefunItem i : radioactiveItems) {
+                    if (i.isItem(tmpItem)) {
+                        exposureTotal += tmpItem.getAmount() * ((RadioactiveItem) i).getRadioactivity().getExposureModifier();
+                    }
+                }
+            }
+        }
+        
+        int exposureLevelBefore = radioactivityLevel.getOrDefault(uuid, 0);
+        if (exposureTotal > 0) {
+            if (exposureLevelBefore == 0) {
+                SlimefunPlugin.getLocalization().sendMessage(p, "messages.radiation");
+            }
+            radioactivityLevel.put(uuid, Math.min(exposureLevelBefore + exposureTotal, 100));
+        } else if (exposureLevelBefore > 0) {
+            radioactivityLevel.put(uuid, Math.max(exposureLevelBefore - 1, 0));
+        }
+        
+        int exposureLevelAfter = radioactivityLevel.getOrDefault(uuid, 0);
+        for (Symptom symptom : SYMPTOMS) {
+            if (symptom.minExposure <= exposureLevelAfter) {
+                applySymptom(symptom, p);
+            }
+        }
+        
+        if (exposureLevelAfter > 0 || exposureLevelBefore > 0) {
+            String msg = SlimefunPlugin.getLocalization().getMessage(p, "actionbar.radiation")
+                    .replace("%level%", "" + exposureLevelAfter);
+            p.spigot().sendMessage(ChatMessageType.ACTION_BAR,
+                    new ComponentBuilder().append(ChatColors.color(msg)).create()
+            );
+        }
+    }
+
+    /**
+     * Applies a symptom to the player.
+     *
+     * @param s Symptom to apply
+     * @param p Player to apply to
+     */
+    private void applySymptom(@Nonnull Symptom s, @Nonnull Player p) {
+        SlimefunPlugin.runSync(() -> {
+            switch (s) {
+                case SLOW: {
+                    p.addPotionEffect(SLOW);
+                    break;
+                }
+                case SLOW_DAMAGE: {
+                    p.addPotionEffect(WITHER);
+                    break;
+                }
+                case BLINDNESS: {
+                    p.addPotionEffect(BLINDNESS);
+                    break;
+                }
+                case FAST_DAMAGE: {
+                    p.addPotionEffect(WITHER2);
+                    break;
+                }
+                case IMMINENT_DEATH: {
+                    p.setHealth(0);
+                    break;
+                }
+            }
+        });
+    }
+
+    private enum Symptom {
```

Comment:
not too sure about putting an enum inside a class

Reply:
not sure about that too, ig its fine

Reply:
Don't put this enum inside a class.